### DISCUSSION TITLE
Help understand nic percentage metrics

### DISCUSSION THREAD

**Discussion by luddite516** [NetApp/harvest#742] [Q&A] [answered]
Reviewing nic_common metrics, I am trying to make sense of the metrics nic_rx_percent and nic_tx_percent, which appear to be calculated by the nic plugin based on link speed and bytes received/sent.

May be best to use an example.  on a 10Gbe link, I have in one case 100MB/sec bytes sent.  This should calculate to about 10% utilization:
    10Gbit at 10/8 link encoding as about 1GByte/sec bandwidth,
    100Mb is about 10% of 1GByte.

But the nic_tx_utilization at this point is .0200.  If I display that as a percent in Grafana, it displays .02%

Please advise.

**Comment by Hardikl** [comment, marked as answer]
So plugin would do this calculation:
1. Fetch tx_bytes and rx_bytes (in bytes)
tx_bytes: 334657
rx_bytes: 21411

![image](https://user-images.githubusercontent.com/83282894/145794895-818467e0-3e7e-4946-9dbf-ec6b060ecdee.png)

![image](https://user-images.githubusercontent.com/83282894/145794908-800356f9-3c45-4fa8-9618-8df0b5e9a8a1.png)

2. Fetch link speed, which is in Mbps from ONTAP and we converted to MBps(bytes per sec) and update the speed label in nic_labels
speed(in bytes): 125000000

![image](https://user-images.githubusercontent.com/83282894/145795065-b14801a2-da13-43cd-a411-5159ad90808e.png)

3. Calculate tx_perc and rx_perc:- both are in same unit
tx_perc: (tx_bytes/speed) 0.002677
rx_perc: (rx_bytes/speed) 0.000017

4. set util percent from max of above
util_perc: max(tx_perc, rx_perc) 0.002677

![image](https://user-images.githubusercontent.com/83282894/145795398-a1d0dd3a-cdc3-43df-80bd-ac90650c1dd3.png)

5. UI would show as 0.26% in the grid 

For your example above, As the rx_bytes and tx_bytes values are changing too frequent, can you please share value of both of them along with speed and UI screenshot at the same time?

Also, I assume you are running on the Harvest 21.11.0 release, please confirm.
